components/script/dom/bindings/str.rs

```diff
@@ -116,6 +116,26 @@ impl ByteString {
             }
         })
     }
+
+    /// Normalize `self`, as defined by
+    /// [the Fetch Spec](https://fetch.spec.whatwg.org/#concept-header-value-normalize).
+    pub fn normalize_header_value(&mut self) {
+        const HTTP_WS_BYTES: &'static [u8] = b"\x09\x0A\x0D\x20";
+
+        let ByteString(ref mut vec) = *self;
+        let back_skip = vec.iter().rev().take_while(|&b| HTTP_WS_BYTES.contains(b)).count();
+
+        if back_skip != 0 {
```

Comment:
`count()` actually *cannot* be smaller than zero since it returns `usize` so `> 0` and `!= 0` are equivalent, in which case it's probably a matter of personal taste  (my preference is `!= 0`, in the same way as I prefer `is_empty()` over `len() > 0`).